src/mappings/mappingUniquesHandlers.ts

```diff
@@ -1,28 +1,156 @@
+  const item = await ensureItem(collectionId.toString(), itemId.toString(), id);
+  item.owner = to.toString();
+  uniqueTransfer.itemId = item.id;
+  uniqueTransfer.collectionId = collection.id;
+
+  await collection.save();
+  await item.save()
   return uniqueTransfer.save();
 }
+
+export const handleUniquesMetadataSetEvent = async (
+  event: SubstrateEvent
+) => {
+  logger.debug("uniqueMetadataSetEvent added: " + JSON.stringify(event.toHuman()));
+  const collectionId = event.event.data[0];
+  const itemId = event.event.data[1];
+  const data = event.event.data[2];
+  const blockNumber = event.block.block.header.number.toNumber();
+  const id = `${blockNumber}-${event.idx}`
+
+  const collection = await ensureCollection(collectionId.toString(), id);
+  const item = await ensureItem(collectionId.toString(), itemId.toString(), id);
+  item.metadataCid = data.toHuman().toString();
+  item.collectionId = collection.id;
+  await collection.save()
+  return item.save();
+}
+
+export const handleUniquesCollectionMetadataSetEvent = async (
+  event: SubstrateEvent
+) => {
+  logger.debug(
+    "uniqueCollectionMetadataSetEvent added: " + JSON.stringify(event.toHuman())
+  );
+  const collectionId = event.event.data[0];
+  const data = event.event.data[1];
+  const blockNumber = event.block.block.header.number.toNumber();
+  const id = `${blockNumber}-${event.idx}`
+
+  const collection = await ensureCollection(collectionId.toString(), id);
+  collection.metadataCid = data.toHuman().toString();
+
+  return collection.save();
+}
+
+export const handleUniquesDestroyedEvent = async (
+  event: SubstrateEvent
+) => {
+  logger.debug(
+    "handleUniquesDestroyedEvent added: " + JSON.stringify(event.toHuman())
+  );
+  const collectionId = event.event.data[0];
+  const blockNumber = event.block.block.header.number.toNumber();
+  const id = `${blockNumber}-${event.idx}`
+
+  const collection = await ensureCollection(collectionId.toString(), id);
+  collection.isDestroyed = true;
+  return collection.save()
+}
+
+export const handleUniquesBurnedEvent = async (
+  event: SubstrateEvent
+) => {
+  logger.debug(
+    "handleUniquesBurnedEvent added: " + JSON.stringify(event.toHuman())
+  );
+  const itemId = event.event.data[1];
+  const collectionId = event.event.data[0];
+  const blockNumber = event.block.block.header.number.toNumber();
+  const id = `${blockNumber}-${event.idx}`
+
+  const item = await ensureItem(collectionId.toString(), itemId.toString(), id);
+  item.isBurned = true;
+  return item.save();
+}
+
+export const handleUniquesIssuedEvent = async (
+  event: SubstrateEvent
+) => {
+  logger.debug(
+    "handleUniquesIssuedEvent added: " + JSON.stringify(event.toHuman())
+  );
+  const collectionId = event.event.data[0];
+  const itemId = event.event.data[1];
+  const owner = event.event.data[2];
+  const blockNumber = event.block.block.header.number.toNumber();
+  const id = `${blockNumber}-${event.idx}`
```

Comment:
To avoid repeating this line in every `handle...Event`, you can pass blockNumber and event.idx to `ensureCollection` and do this inside those functions. This way we can ensure the id is always created following the same pattern.

Reply:
I've update the ensureCollection and ensureItem arguments